lib/robot.js

```diff
@@ -1,28 +1,22 @@
-const {EventEmitter} = require('promise-events')
 const GitHubApi = require('github')
 const Bottleneck = require('bottleneck')
 const express = require('express')
-const Context = require('./context')
 
 /**
  * The `robot` parameter available to apps
  *
  * @property {logger} log - A logger
  */
 class Robot {
-  constructor ({app, cache, logger, router, catchErrors} = {}) {
-    this.events = new EventEmitter()
+  constructor ({app, cache, logger, webhooks: {on, removeListener, handle} = {}, router, catchErrors} = {}) {
     this.app = app
     this.cache = cache
     this.router = router || new express.Router()
     this.log = wrapLogger(logger)
     this.catchErrors = catchErrors
-  }
-
-  async receive (event) {
-    return this.events.emit('*', event).then(() => {
-      return this.events.emit(event.event, event)
-    })
+    this.on = on
+    this.removeListener = removeListener
+    this.receive = handle
```

Comment:
I don't want to pass the webhook handler directly into robot because someday soon, this event emitter will likely used for more than just GitHub webhooks (#265).

Reply:
with #265 I guess we will have a namespaced `.receive` method, like 

```
receive (namespace, event) {}
```

Where `namespace` would be `github`, `slack`, etc?

That will be a breaking change anyway, shall we merge this in as is and address #265 once we get there?

Reply:
note that the existing `receive` method expects {event, payload}. The `receive` method coming from `@octokit/webhooks` expects `{id, name, data}`, so it’s a breaking change.

I’ll define wrapper `receive` methods in `/index.js` and `/lib/robot.js` to match the current options. 

I’d suggest log a `deprecation` if `{event, payload}` is passed so folks migrate to `{id, name, data}` in their tests, without shipping a breaking change. What do you think?